airbyte-integrations/connectors/source-mssql/src/test-performance/java/io/airbyte/integrations/source/mssql/MsSqlRdsSourcePerformanceSecretTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mssql;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.standardtest.source.TestDestinationEnv;
+import io.airbyte.integrations.standardtest.source.performancetest.AbstractSourcePerformanceTest;
+import java.nio.file.Path;
+import org.junit.jupiter.api.Test;
+
+public class MsSqlRdsSourcePerformanceSecretTest extends AbstractSourcePerformanceTest {
+
+  private JsonNode config;
+  private static final String PERFORMANCE_SECRET_CREDS = "secrets/performance-config.json";
+
+  @Override
+  protected JsonNode getConfig() {
+    return config;
+  }
+
+  @Override
+  protected void tearDown(final TestDestinationEnv testEnv) {}
+
+  @Override
+  protected String getImageName() {
+    return "airbyte/source-mssql:dev";
+  }
+
+  @Override
+  protected void setupDatabase(String dbName) {
+    JsonNode plainConfig = Jsons.deserialize(IOs.readFile(Path.of(PERFORMANCE_SECRET_CREDS)));
+
+    config = Jsons.jsonNode(ImmutableMap.builder()
+        .put("host", plainConfig.get("host"))
+        .put("port", plainConfig.get("port"))
+        .put("database", dbName)
+        .put("username", plainConfig.get("username"))
+        .put("password", plainConfig.get("password"))
+        .build());
+  }
+
+  @Test
+  public void test100tables100recordsDb() throws Exception {
+    int numberOfDummyRecords = 100;
+    int numberOfColumns = 240;
+    int numberOfStreams = 100;
+    String defaultDbSchemaName = "dbo";
+
+    setupDatabase("test100tables100recordsDb");
+
+    performTest(defaultDbSchemaName, numberOfStreams, numberOfColumns, numberOfDummyRecords);
+  }
```

Comment:
All these test methods are the same / almost the same across the three databases. Can we move them to the upstream `AbstractSourcePerformanceTest` class as well? In that way, it is much simpler to add or remove a test.

Reply:
Updated, thanks!